contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -166,22 +167,22 @@ pub fn try_refund<S: Storage, A: Api, Q: Querier>(
     let msgs = send_tokens(&deps.api, &env.contract.address, &rcpt, swap.balance)?;
     Ok(HandleResponse {
         messages: msgs,
-        log: vec![log("action", "refund"), log("id", id), log("to", rcpt)],
+        attributes: vec![attr("action", "refund"), attr("id", id), attr("to", rcpt)],
         data: None,
     })
 }
 
-fn parse_hex_32(data: &str) -> StdResult<Vec<u8>> {
+fn parse_hex_32(data: &str) -> Result<Vec<u8>, ContractError> {
     match hex::decode(data) {
         Ok(bin) => {
             if bin.len() == 32 {
                 Ok(bin)
             } else {
-                Err(StdError::generic_err("Hash must be 64 characters"))
+                Err(ContractError::Invalid("hash: must be 64 characters".into()))
             }
         }
-        Err(e) => Err(StdError::generic_err(format!(
-            "Error parsing hash: {}",
+        Err(e) => Err(ContractError::ParseError(format!(
```

Comment:
You can also give some text to the error itself rather than combine messages here.

`ContractError::ParseHash(e.to_string())`

```
   #[error("non-hex hash: {0}")]
    ParseHashError(String),
```

I generally prefer avoiding the `format!()` directive when making errors. I used that a lot in generic error, but if we have custom types, let's worry about the display in `errors.rs` and just pass in needed context here (if any)

Reply:
Yes. I'll move all the formatting to the error itself, as mentioned before.